### ISSUE TITLE
Make mandatory settings part of struct instead of being options

### ISSUE THREAD

**Issue by derrandz** [celestiaorg/celestia-node#1905]
## Overview

Context: https://github.com/celestiaorg/celestia-node/pull/1798#discussion_r1133712856

**Comment by renaynay** [comment]
I agree with this approach but it's very low priority at the moment.

Let's make it a good first issue.
